packages/web-api/src/methods.ts

```diff
@@ -1593,25 +1641,174 @@ export interface ChatPostEphemeralArguments extends WebAPICallOptions, TokenOver
+
+interface ChatPostMessageArgumentsIdentityBase {
+  /**
+   * @description (Legacy) Pass `true` to post the message as the authed user instead of as a bot. Defaults to `false`.
+   * Can only be used by classic Slack apps.
+   * @see {@link https://api.slack.com/methods/chat.postMessage#legacy_authorship Legacy Concerns and Authorship when using `chat.postMessage`}.
+   */
+  as_user?: boolean; // Legacy
+}
+
+/**
+ * @description `chat.postMessage` arguments for specifying the message icon using a URL.
+ */
+interface ChatPostMessageArgumentsIconURL extends ChatPostMessageArgumentsIdentityBase {
+  as_user: false;
+  /**
+   * @description URL to an image to use as the icon for this message.
+   */
+  icon_url: string;
+}
+
+/**
+ * @description `chat.postMessage` arguments for specifying the message icon using an emoji.
+ */
+interface ChatPostMessageArgumentsIconEmoji extends ChatPostMessageArgumentsIdentityBase {
+  as_user: false;
+  /**
+   * @description Emoji to use as the icon for this message. Overrides `icon_url`. E.g. `:chart_with_upwards_trend:`
+   */
+  icon_emoji: string;
+}
+
+/**
+ * @description `chat.postMessage` arguments for specifying the message username.
+ */
+interface ChatPostMessageArgumentsUsername extends ChatPostMessageArgumentsIdentityBase {
+  as_user: false;
+  /**
+   * @description Set your bot's user name.
+   */
+  username: string;
+}
+
+type ChatPostMessageArgumentsIdentity = ChatPostMessageArgumentsIdentityBase | ChatPostMessageArgumentsIconEmoji |
+ChatPostMessageArgumentsIconURL | ChatPostMessageArgumentsUsername;
+
+/**
+ * @description `chat.postMessage` arguments for specifying a thread reply, with optional conversation broadcast.
+ */
+interface ChatPostMessageArgumentsThreadReply {
+  thread_ts: string;
+  reply_broadcast?: boolean;
+}
+
+/**
+ * @description `chat.postMessage` arguments for specifying a 'normal' message - not a thread reply - which will never
+ * have a `thread_ts`.
+ */
+interface ChatPostMessageArgumentsConversationMessage {
+  thread_ts?: never;
+  reply_broadcast?: never;
+}
+
+type ChatPostMessageArgumentsType = ChatPostMessageArgumentsThreadReply | ChatPostMessageArgumentsConversationMessage;
+
+type ChatPostMessageArguments =
+  // Must provide one of the following to determine message contents
+  ChatPostMessageArgumentsContents &
+  // Controls the username/identity and icon display for the message
+  ChatPostMessageArgumentsIdentity &
+  // Controls whether a message is a thread reply or not
+  ChatPostMessageArgumentsType;
+
+const textOnly: ChatPostMessageArguments = {
+  channel: 'C1234',
+  text: 'hi',
+};
+const attachments: ChatPostMessageArguments = {
+  channel: 'C1234',
+  attachments: [{}],
+  text: 'some fallback',
+};
+const bs: ChatPostMessageArguments = {
+  channel: 'C1234',
+  blocks: [{
+    type: '',
+  }],
+  thread_ts: '123',
+  reply_broadcast: true,
+  icon_url: '1234',
+};
+console.log(textOnly, attachments, bs);
```

Comment:
TODO: remove this (perhps the tests in this file too)

Reply:
Yes exactly, this is just a draft to have a discussion.